src/controllers/dynakube/istio/serviceentry.go

```diff
@@ -60,3 +71,44 @@ func buildServiceEntryIP(name, namespace, host string, port uint32) *istiov1alph
 		},
 	}
 }
+
+func handleIstioConfigurationForServiceEntry(instance *dynatracev1beta1.DynaKube,
+	name string, communicationHost dtclient.CommunicationHost, role string, config *rest.Config, namespace string,
+	istioClient istioclientset.Interface, scheme *runtime.Scheme, log logr.Logger) (bool, error) {
+
+	probe, err := kubeobjects.KubernetesObjectProbe(ServiceEntryGVK, instance.GetNamespace(), name, config)
```

Comment:
IMO, its fine here, as its a utility function so creating a separate struct wouldn't be that beneficial

Reply:
I could export the suggested struct which is needed in the `handleIstioConfigurationForVirtualService() `method for example and pass it here as well. What do you guys think about that, or should `KubernetesObjectProbe() `stay as it is?

Reply:
`KubernetesObjectProbe ` should be independent from the istio package if its going to be in the `kubeobjects` package. (if you exported in the istio package and used in the kubeobjects packaged you would get import circle which is bad)

having a guideline of `no more then 2 params for a function` is nice and all, but that is too unrealistic and fragile to enforce seriously IMO. 
Because:
- having random single use structs are worse then a 3 or 4 param function.
- to couple 2 supposedly independent things with a struct, because the required params would match whats in the struct is not a good idea. You just made 2 loosely coupled things tightly coupled for no good reason.

Reply:
in my opinion you stated very good points, on why it should stay the same. Thank you guys. (and as you mentioned, the import cycle would cause problems anyways)

Reply:
`KubernetesObjectProbe`  can be made independent from the istio package if the struct is part of the `kubeobjects` package and not of the istio package. Furthermore, having multiple params is worse than having single use structs, because structs encapsulate data into a group which makes code better understandable.

In this case, for example, the paramter list `ServiceEntryGVK, instance.GetNamespace(), name, config` does not say much. However, if you bundle it in a struct it is much more understandable.
E.g.:
```
type KubernetesObjectProbe struct {
  serviceEntry string
  instanceName string
  name string
  config *rest.Config
}
```
The data is now grouped together.
Adding a function, now without parameters, allows for an easy to read call for a result.
```
func (probe *KubernetesObjectProbe) GetProbeResult() (ProbeResult, error) {... }
```

If you need to use it, the code would then look like so:
```
objectProbe := &kubeobjects.KubernetesObjectProbe {
 ...
}
probeResult, err := objectProbe.GetProbeResult()
```

It is more lines of code, but better read- and understandable

Reply:
I agree to disagree :D

Using a struct like so would kinda warrant a constructor, which would result in a function with 4 params in this case 😬 
so we didn't really achieve anything just more code to maintain

Reply:
🙂

Reply:
Reasons to avoid long parameter lists
https://refactoring.guru/smells/long-parameter-list
https://www.arhohuttunen.com/long-parameter-list/
https://medium.com/@huytrongnguyen1985/lessons-learnt-from-the-clean-code-robert-c-martin-cecbe2b09139

If you are worried about long constructors, use the builder pattern
https://konfhub.medium.com/refactoring-long-parameter-list-in-constructors-java-e9c9f2cb1fb